pkg/kubelet/kubelet.go

```diff
@@ -170,16 +170,30 @@ func (kl *Kubelet) GetRootDir() string {
 
 // GetPodsDir returns the full path to the directory under which pod
 // directories are created.
-// TODO(thockin): For now, this is the same as the root because that is assumed
-// in other code.  Will fix.
 func (kl *Kubelet) GetPodsDir() string {
-	return kl.GetRootDir()
+	return path.Join(kl.GetRootDir(), "pods")
 }
 
 // GetPodDir returns the full path to the per-pod data directory for the
 // specified pod.  This directory may not exist if the pod does not exist.
 func (kl *Kubelet) GetPodDir(podUID string) string {
```

Comment:
I worry about security here.  A sanity check that there are no special characters here would help.  This is the type of code that can often lead to a directory traversal attack.